v2/api/authorization/customizations/role_assignment_extensions.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package customizations
+
+import (
+	"context"
+	"strings"
+
+	api "github.com/Azure/azure-service-operator/v2/api/authorization/v1api20200801preview"
+
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/extensions"
+)
+
+var _ extensions.Importer = &RoleAssignmentExtension{}
+
+func (extension *RoleAssignmentExtension) Import(
+	ctx context.Context,
+	rsrc genruntime.ImportableResource,
+	owner *genruntime.ResourceReference,
```

Comment:
An alternative to passing `owner` here would be, if there's a way for each resource to get its own ARM ID, you could use the ParseARMID function of the Azure SDK to decompose the ARM ID and then determine its owner from that.

I don't know if you have such a way to get the resources ARM ID already, but if you do it would save you from needing to modify your interface (which might be cleaner?)

Not critical though if there is no way for the resource to get its own ARM ID already.

Reply:
I considered doing that - but realized it wouldn't work for non-ARM resources (such as database users).